From 470cf490d95b01ec25b5595b01c98e145a935726 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Tue, 20 Jun 2017 16:33:55 +0300
Subject: [PATCH 1983/5242] MLK-15067: ASoC: fsl: imx-wm8958: Kill warning for
 non-gpr boards

commit  b8458944e02eb9276ebe2d0a3e0488f365cc3d32 from
https://source.codeaurora.org/external/imx/linux-imx.git

Similar with 7c280619ed45b (" MLK-14663-2: ASoC: fsl: imx-wm8960: Kill
warning for non-gpr boards")

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8958.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index 707cf59..a8343d1 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -407,6 +407,10 @@ static int of_parse_gpr(struct platform_device *pdev,
 	int ret;
 	struct of_phandle_args args;
 
+	if (of_device_is_compatible(pdev->dev.of_node,
+				    "fsl,imx7d-12x12-lpddr3-arm2-wm8958"))
+		return 0;
+
 	ret = of_parse_phandle_with_fixed_args(pdev->dev.of_node,
 					       "gpr", 3, 0, &args);
 	if (ret) {
@@ -561,6 +565,7 @@ static int imx_wm8958_remove(struct platform_device *pdev)
 
 static const struct of_device_id imx_wm8958_dt_ids[] = {
 	{ .compatible = "fsl,imx-audio-wm8958", },
+	{ .compatible = "fsl,imx7d-12x12-lpddr3-arm2-wm8958", },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, imx_wm8958_dt_ids);
-- 
1.7.9.5

